R/ab_scenario.R

```diff
@@ -18,122 +20,90 @@
-#'   leeds_desire_lines,
-#'   leeds_zones,
-#'   scenario = "dutch",
-#'   output_format = "json_list",
-#'   hr = 20, # representing 8 pm
-#'   sd = 0
-#' )
-#' f = tempfile(fileext = ".json")
-#' ab_save(ab_evening_dutch, f)
-#' readLines(f)[13]
-#' 20 * 60^2
-ab_scenario = function(houses,
-                       buildings,
-                       desire_lines,
-                       zones,
-                       scenario = "base",
-                       time_fun = ab_time_normal,
-                       output_format = "sf",
-                       op = sf::st_intersects,
-                       ...
-                       ) {
-
-  requireNamespace("sf", quietly = TRUE)
-  # loop over each desire line
-  i = 1
-  for(i in seq(nrow(desire_lines))) {
-    # todo: allow the user to choose the 'pop var' or document (RL 2020-02-10)
-    pop = desire_lines$all_base[i]
-    cnames = names(desire_lines)
-
-    # todo: split out the lines to odc_to_sf as function? (RL 2020-02-10)
-    # supressing messages associated with GEOS ops on unprojected data
-    suppressWarnings({
-      suppressMessages({
-        origins = houses[sample(nrow(houses), size = pop, replace = TRUE), ]
-        destination_zone = zones %>% subset(geo_code == desire_lines$geo_code2[i])
-        destination_buildings = buildings[destination_zone, , op = op]
-        destinations = destination_buildings[sample(nrow(destination_buildings), size = pop, replace = TRUE), ]
-        origin_coords = origins %>% sf::st_centroid() %>% sf::st_coordinates()
-        destination_coords = destinations %>% sf::st_centroid() %>% sf::st_coordinates()
-        desire_lines_disag = od::odc_to_sf(odc = cbind(origin_coords, destination_coords))
-      })
+#' plot(ablines["mode"], add = TRUE)
+#' ablines_json = ab_json(ablines)
+#' # ablines = ab_scenario(
+#' #   leeds_houses,
+#' #   leeds_buildings,
+#' #   leeds_desire_lines,
+#' #   leeds_zones,
+#' #   output_format = "sf"
+#' # )
+#' # dutch = ab_scenario(
+#' #   leeds_houses,
+#' #   leeds_buildings,
+#' #   leeds_desire_lines,
+#' #   leeds_zones,
+#' #   scenario = "dutch",
+#' #   output_format = "sf"
+#' # )
+#' # plot(ablines, key.pos = 1, reset = FALSE)
+#' # plot(leeds_site_area$geometry, add = TRUE)
+#' # plot(leeds_buildings$geometry, add = TRUE)
+#' # plot(dutch, key.pos = 1, reset = FALSE)
+#' # plot(leeds_site_area$geometry, add = TRUE)
+#' # plot(leeds_buildings$geometry, add = TRUE)
+#' # dutch$departure = ab_time_normal(hr = 8, sd = 0.5, n = nrow(dutch))
+#' # ab_evening_dutch = ab_scenario(
+#' #   leeds_houses,
+#' #   leeds_buildings,
+#' #   leeds_desire_lines,
+#' #   leeds_zones,
+#' #   scenario = "dutch",
+#' #   output_format = "json_list",
+#' #   hr = 20, # representing 8 pm
+#' #   sd = 0
+#' # )
+#' # f = tempfile(fileext = ".json")
+#' # ab_save(ab_evening_dutch, f)
+#' # readLines(f)[13]
+#' # 20 * 60^2
+ab_scenario = function(
+  od,
+  zones,
+  zones_d = NULL,
+  subpoints = NULL,
+  subpoints_d = NULL,
```

Comment:
These params don't match the docs.. I'm assuming `subpoints_d` is `buildings`?

Reply:
There is a questionmark around what to call these params. Initially I called them `houses` and `buildings` but, because this new implementation uses `od_jitter()` I switched to be faithful to that function. If we go with this draft PR the parameter names in the docs would also be updated. Issue with `houses` and `buildings`: they are rather specific, so `subpoints` could be more generalisable. But yes, exactly, that is what's going on.

Reply:
`origin_buildings` and `destination_buildings` are clear to me, personally. The term "subpoint" from a geometric perspective is confusing, since a point shouldn't have any constituent parts. :P

Reply:
> `origin_buildings` and `destination_buildings` are clear to me, personally. The term "subpoint" from a geometric perspective is confusing, since a point shouldn't have any constituent parts. :P

Good feedback, will go with that. The good thing about working on an alpha-state package that few people are using: you can shift things around and make breaking changes without too much pain. On that note: I could bump this to v. 1 but I still don't think it deserves that name. This will be a major change though and the code in ActDev will have to use a pinned version of this package, e.g. with

```
remotes::install_github("a-b-street/abstr", ref = "commit_hash_before_merge")
```
or we'll update the code (recommended, this approach is more computationally efficient I think).

Reply:
Done.